web/src/components/layout/layout.tsx

```diff
@@ -162,6 +166,17 @@ class Layout extends React.PureComponent<LayoutProps, LayoutState> {
 
     return (
       <div id="main" className={className}>
+        {showWelcomeModal && (
+          <WelcomeModal
+            onRequestClose={() => {
+              this.setState({ showWelcomeModal: false });
+            }}
+            onClick={() => {
+              window.location.pathname = '/login';
```

Comment:
I went through several iterations of fancy react-router magic before landing on this. If you have any examples of a cleaner redirect to uncontrolled routes, please let me know!

Also: we probably need to pass information with this redirect, to ensure the extra Open Voice Challenge fields (eg. `team`) get populated correctly. We're working on a solution as the backend comes together, but any input is appreciated.

Reply:
Oh I'd go with a regular ol' `<a>`. We do pass some state around for login (namely locale and when changing the email the old_email) here: https://github.com/mozilla/voice-web/blob/master/server/src/auth-router.ts#L130-L140

Hope that helps!

Reply:
@Gregoor the button needs to be disabled until the terms & conditions box is checked; how would you recommend handling the disabled state? I thought there may be some a11y concerns with trying to fake a "disabled" link by changing the href, but yeah, I'm not happy with the button solution either… 😕 If you know of anywhere else on the site where this pattern is followed I'd appreciate a link!

Reply:
`aria-disabled` could work to convey state, swapping button for link seems not so great as I suspect it would unexpectedly change semantics (unexpectedness can affect certain users disproportionately)

Reply:
Great, thankslots for the guidance!